### DISCUSSION TITLE
Alias to trigger URL

### DISCUSSION THREAD

**Discussion by svoop** [Mailu/Mailu#3035] [Feature requests / Ideas]
On my old, cumbersome vanilla Postfix install, I had a few aliases which forwarded to `|/opt/bin/whatever.sh`. One use case for this: A "secret" email address like `alarm123@example.com` which triggered a script so send an SMS to my mobile phone. (My provider kindly implements a GET-request to SMS gateway.)

Now, on Mailu, things are a little more complicated. I guess it's somehow possible to override the aliases file, but maybe it would be a nice extension of Mailu to allow for URLs as alias destination along with some placeholders. Say:

```
https://api.example.com/inbound?subject=%s&from=%f
                                                ^^ URL encoded FROM
                                        ^^ URL encoded SUBJECT
```

As a side note: If I remember correctly, per-domain forwards are part of `virtual`, but piping to a CLI command is only possible in `aliases`. On my vanilla box, I had something along the lines of `alarm123@example.com --virtual--> alarm123_example_com --aliases--> |/opt/bin/whatever.sh`.

I could provide a shell script which builds and does the actual request, should be possible with `curl` etc already available in the containers.

What do you think?

**Comment by nextgens** [comment]
I think you can already do this using sieve.

Create ``overrides/dovecot/myscript.script`` ... with something like:
```
#!/bin/sh
curl ...
```
When the imap container restarts it will be deployed [in the right place](https://github.com/Mailu/Mailu/blob/master/core/dovecot/start.py#L27)

Then define a sieve script with ``require "vnd.dovecot.execute";execute :pipe myscript;``.

https://github.com/dovecot/pigeonhole/blob/main/doc/rfc/spec-bosch-sieve-extprograms.txt and https://doc.dovecot.org/configuration_manual/sieve/plugins/extprograms/#example-1 are a must read if you want to do fancier.

We have #1504 already for tracking this
